src/inventory_ui.cpp

```diff
@@ -990,6 +1121,180 @@ item_location game::inv_map_splice(
     }
 }
 
+item_location game::inv_map_splice_adjacent( item_filter inv_filter, item_filter ground_filter, item_filter vehicle_filter,
```

Comment:
This looks like it's a copy of `game::inv_map_splice`, what has it to do with "read book on ground" (the PR title).

It's probably a good feature, but it should extend the existing `game::inv_map_splice` function, this doesn't require a separate function.

And it should probably go into its own PR.